src/assets/scripts/client/airport/runway/RunwayCollection.js

```diff
@@ -0,0 +1,264 @@
+
+/**
+ * Collection of `RunwayModel`s
+ *
+ * Provides methods to reason about the various `RunwayModel`s belonging to an airport
+ *
+ * @class RunwayCollection
+ * @extends BaseCollection
+ */
+export default class RunwayCollection extends BaseCollection {
+    /**
+     * @constructor
+     * @param runwayJson {array<object>}
+     * @param airportPositionModel {StaticPositionModel}
+     */
+    constructor(runwayJson, airportPositionModel) {
+        super();
+
+        if (!_isArray(runwayJson)) {
+            throw new TypeError(
+                `Invalid parameter passed to RunwayCollection. Expected an array but found ${typeof runwayJson}`
+            );
+        }
+
+        /**
+         * @inherited
+         * @memberof BaseCollection
+         * @property _items
+         * @type {array<RunwayModel>}
+         * @default []
+         */
+
+         /**
+          * @inherited
+          * @memberof BaseCollection
+          * @property length
+          * @type {number}
+          * @default #_items.length
+          */
+
+        /**
+         * @property _airportPositionModel
+         * @type {StaticPositionModel}
+         * @default null
+         * @private
+         */
+        this._airportPositionModel = null;
+
+        /**
+         * Dictionary of all airport runways, organized by runway name, that
+         * contains a `RunwayRelationshipModel` for each runway
+         *
+         * @property _runwayRelationships
+         * @type object
+         * @default {}
+         * @private
+         */
+        this._runwayRelationships = {};
+
+        this._init(runwayJson, airportPositionModel);
+    }
+
+    /**
+     * Public fascade for `#_items`
+     *
+     * @property runways
+     * @return {array<RunwayModel>}
+     */
+    get runways() {
+        return this._items;
+    }
+
+    /**
+     * Initialize the instance
+     *
+     * @for RunwayCollection
+     * @method _init
+     * @param runwayJson {array<object>}
+     * @param airportPositionModel {StaticPositionModel}
+     */
+    _init(runwayJson, airportPositionModel) {
+        this._airportPositionModel = airportPositionModel;
+
+        this._buildRunwayModels(runwayJson);
+        this._buildRunwayRelationships();
+    }
+
+    /**
+     * Tear down the instance and destroy any class property values
+     *
+     * @for RunwayCollection
+     * @method destroy
+     */
+    destroy() {
+        this._airportPositionModel = null;
+        this._runwayRelationships = {};
+    }
+
+    /**
+     * Based on the current wind, find the most appropriate runway for use
+     *
+     * Used to find a default runway
+     *
+     * @for findBestRunwayForWind
+     * @method findBestRunwayForWind
+     * @param getCurrentWindProps {function<object>}
+     * @return bestRunway {string}
+     */
+    findBestRunwayForWind(getCurrentWindProps) {
+        // FIXME: figure out what this does and move it to a more appropriate home
+        const ra = (n) => {
```

Comment:
@erikquinn I still don;t quite understand why we need this and why a random number is used in the calculations. 😞